mods/toggle-taskbar-autohide.wh.cpp

```diff
@@ -0,0 +1,1108 @@
+            g_taskbarVersion = WIN_10_TASKBAR;
+        } else {
+            g_taskbarVersion = WIN_11_TASKBAR;
+        }
+    } else {
+        g_taskbarVersion = UNKNOWN_TASKBAR;
+    }
+
+    return TRUE;
+}
+
+void LoadSettings() {
+    g_settings.oldTaskbarOnWin11 = Wh_GetIntSetting(L"oldTaskbarOnWin11");
+}
+
+bool GetTaskbarAutohideState() {
+    if (g_hTaskbarWnd == NULL) {
+        return false;
+    }
+    APPBARDATA msgData{};
+    msgData.cbSize = sizeof(msgData);
+    msgData.hWnd = g_hTaskbarWnd;
+    LPARAM state = SHAppBarMessage(ABM_GETSTATE, &msgData);
+    return state & ABS_AUTOHIDE;
+}
+
+void SetTaskbarAutohide(bool enabled) {
+    if (g_hTaskbarWnd == NULL) {
+        return;
+    }
+
+    APPBARDATA msgData{};
+    msgData.cbSize = sizeof(msgData);
+    msgData.hWnd = g_hTaskbarWnd;
+    msgData.lParam = enabled ? ABS_AUTOHIDE : ABS_ALWAYSONTOP;
+    SHAppBarMessage(ABM_SETSTATE, &msgData);
+}
+
+bool ToggleTaskbarAutohide() {
+    const bool isEnabled = GetTaskbarAutohideState();
+    SetTaskbarAutohide(!isEnabled);
+    return !isEnabled;
+}
+
+#pragma endregion // functions
+
+// =====================================================================
+
+// main body of the mod called every time a taskbar is clicked
+bool OnMouseClick(HWND hWnd, WPARAM wParam, LPARAM lParam) {
+    if (GetCapture() != NULL) {
+        return false;
+    }
+
+    // old Windows mouse handling of WM_MBUTTONDOWN message
+    POINT pointerLocation{};
+    if (g_taskbarVersion == WIN_10_TASKBAR) {
+        // message carries mouse position relative to the client window so use GetCursorPos() instead
+        if (!GetCursorPos(&pointerLocation)) {
+            Wh_Log(L"Failed to get mouse position");
+            return false;
+        }
+
+    // new Windows sends different message - WM_POINTERDOWN
+    } else {
+        if (IS_POINTER_THIRDBUTTON_WPARAM(wParam) == 0) {
+            return false;
+        }
+        POINTER_INFO pointerInfo{};
+        BOOL success = GetPointerInfo(GET_POINTERID_WPARAM(wParam), &pointerInfo);
+        if (!success) {
+            return false;
+        }
+        pointerLocation = pointerInfo.ptPixelLocation;
+    }
+    Wh_Log(L"Middle mouse clicked at x=%ld, y=%ld", pointerLocation.x, pointerLocation.y);
+
+    // Note: The reason why UIAutomation interface is used is that it reliably returns a className of the element clicked.
+    // If standard Windows API is used, the className returned is always Shell_TrayWnd which is a parrent window wrapping the taskbar.
+    // From that we can't really tell reliably whether user clicked on the taskbar empty space or on some UI element on that taskbar, like 
+    // opened window, icon, start menu, etc. 
+
+    IUIAutomation *pUIAutomation = g_UIAutomation.getInstance();
+    if (pUIAutomation == NULL) {
+        Wh_Log(L"UIAutomationWrapper failed to get instance\n");
+        return false;
+    }
+
+    IUIAutomationElement *pWindowElement = NULL;
```

Comment:
FYI you can use COM smart pointers to avoid having to call `Release`. Example:
https://github.com/ramensoftware/windhawk-mods/blob/90d6ccd363cb9764f081ea1981932b2b3dac68cf/mods/taskbar-grouping.wh.cpp#L726-L729